data/account/participationRegistry.go

```diff
@@ -763,13 +789,8 @@ func (db *participationDB) GetStateProofForRound(id ParticipationID, round basic
 
 		return nil
 	})
-	switch err {
-	case nil:
-		// no error, continue
-	case ErrSecretNotFound: // not considered an error (yet), since some accounts may not have registered state proof yet
-		return result, nil
-	default:
-		return StateProofRecordForRound{}, err
+	if err != nil {
+		return StateProofRecordForRound{}, fmt.Errorf("failed to fetch state proof for round %d: %w", round, err)
```

Comment:
It should be more clear that these are state proof key secrets, not a "state proof". Similarly this function's name probably should be GetStateProofRecordForRound so it is more clear this function gets records/secrets and not proofs.